src/frequenz/sdk/timeseries/_quantities.py

```diff
@@ -0,0 +1,310 @@
+                )
+            if fspec_parts[0] == "0":
+                keep_trailing_zeros = True
+            precision = int(fspec_parts[1])
+        else:
+            precision = 3
+        abs_value = abs(self._base_value)
+        exponent = math.floor(math.log10(abs_value))
+        unit_place = exponent - exponent % 3
+        if unit_place < min(self._exponent_unit_map):
+            unit = self._exponent_unit_map[min(self._exponent_unit_map.keys())]
+            unit_place = min(self._exponent_unit_map)
+        elif unit_place > max(self._exponent_unit_map):
+            unit = self._exponent_unit_map[max(self._exponent_unit_map.keys())]
+            unit_place = max(self._exponent_unit_map)
+        else:
+            unit = self._exponent_unit_map[unit_place]
+        value_str = f"{self._base_value / 10 ** unit_place:.{precision}f}"
+        stripped = value_str.rstrip("0").rstrip(".")
+        if not keep_trailing_zeros:
+            value_str = stripped
+        unit_str = unit if stripped != "0" else self._exponent_unit_map[0]
+        return f"{value_str} {unit_str}"
+
+    def __gt__(self, other: Self) -> bool:
+        """Return whether this quantity is greater than another.
+
+        Args:
+            other: The other quantity.
+
+        Returns:
+            Whether this quantity is greater than another.
+        """
+        return self._base_value > other._base_value
+
+    def __ge__(self, other: Self) -> bool:
+        """Return whether this quantity is greater than or equal to another.
+
+        Args:
+            other: The other quantity.
+
+        Returns:
+            Whether this quantity is greater than or equal to another.
+        """
+        return self._base_value >= other._base_value
+
+    def __lt__(self, other: Self) -> bool:
+        """Return whether this quantity is less than another.
+
+        Args:
+            other: The other quantity.
+
+        Returns:
+            Whether this quantity is less than another.
+        """
+        return self._base_value < other._base_value
+
+    def __le__(self, other: Self) -> bool:
+        """Return whether this quantity is less than or equal to another.
+
+        Args:
+            other: The other quantity.
+
+        Returns:
+            Whether this quantity is less than or equal to another.
+        """
+        return self._base_value <= other._base_value
+
+    def __eq__(self, other: object) -> bool:
+        """Return whether this quantity is equal to another.
+
+        Args:
+            other: The other quantity.
+
+        Returns:
+            Whether this quantity is equal to another.
+        """
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+        return self._base_value == other._base_value
+
+
+class Power(
+    Quantity,
+    exponent_unit_map={
+        -3: "mW",
+        0: "W",
+        3: "kW",
+        6: "MW",
+    },
```

Comment:
I tried to keep only the commonly used units,  so that it doesn't get confusing.  For current, for example, `kA` feels unfamiliar.

But maybe you're right.  Strict adherence to the standards doesn't sound so bad.  I'll make the change.

Reply:
Although might not be so easy to make general, because of types like `°C` for temperature.

Reply:
> I'll make the change.

:+1: Just a few more comments for completeness...

> I tried to keep only the commonly used units, so that it doesn't get confusing. For current, for example, kA feels unfamiliar.

Yeah, but the thing is that should never happen unless something is really wrong. We can still not provide `from_xxx()` and `as_xxx()` methods that are not natural, this will only be used for printing, and if for some weird reason you are working with 2389484A, I think it is still more legible to have 2.38 MA than 2389484 A, even if the MA looks weird, because it is pretty damn weird to have 2.38 MA. So IMHO we gain in safety (people can't add a map with some error), DRY (people don't need to create a new map over and over again for each unit) and predictability (all units use the same system).

> Although might not be so easy to make general, because of types like °C for temperature.

What about `°C`? You mean we should use Kelvin because that's the base SI unit or just that `K°C` also looks weird? If it is the later, IMHO is the same as I say above, that should never happen because otherwise we have burning batteries, which is a bigger problem that weirdness while reading logs :laughing: . If you mean the later, then :grimacing: . I guess we can make an exception and not use the SI unit for that.

Reply:
Also this doesn't work for things like area, volume, etc,  because 1 sq.km is 1000000 sq.m.

Reply:
And we're not using SI units everywhere,  for example energy is J,  but we use kWh,  which simplifies to a weird J/s*s. :D

Reply:
https://github.com/frequenz-floss/frequenz-sdk-python/issues/449